appserver/extras/payara-micro/payara-micro-boot/src/main/java/fish/payara/micro/PayaraMicroRuntime.java

```diff
@@ -138,7 +139,7 @@
      * @param args The parameters to the command
      * @return
      */
-    public Map<InstanceDescriptor, Future<? extends ClusterCommandResult>> run(Collection<InstanceDescriptor> members, String command, String... args);
+    public Map<InstanceDescriptor, Future<? extends CommandResult>> run(Collection<InstanceDescriptor> members, String command, String... args);
```

Comment:
I'd suggest solving the compatibility problem here by solving ambiguity of the overload of `run` as well -- running admin commands, and running Hazelcast batch.

Mark these two methods as deprecated, while keeping their signature.

Add `executeAdminCommand` methods (or `runAdminCommand` or similar) with new signature.